diff -ruN -x '*.o' -x '*.ko' -x '*.mod*' -x '.tmp*' -x Module.symvers -x modules.order a/fantsrvkm/fantdpu_dp_debugfs.c b/fantsrvkm/fantdpu_dp_debugfs.c
--- a/fantsrvkm/fantdpu_dp_debugfs.c	2022-01-01 08:00:00.000000000 +0800
+++ b/fantsrvkm/fantdpu_dp_debugfs.c	2026-09-10 09:08:41.540625263 +0800
@@ -559,13 +559,15 @@
 	}
 
 	backlight_mode = fantdpu_get_connector_backlight_mode(dp_dev->dev, CONNECTOR_M_DP0 + dp_dev->dp_id);
-	if (backlight_mode >= CONNECTOR_BACKLIGHT_PWM0 &&
+	if (backlight_mode >= CONNECTOR_BACKLIGHT_DDCCI &&
 		backlight_mode <= CONNECTOR_BACKLIGHT_AUX_HDR_BL) {
 		dp_dev->panel = panel_pwr_create(connector->dev, dp_dev->dev,
 										  CONNECTOR_M_DP0 + dp_dev->dp_id,
 										  &dp_dev->aux,
 										  connector);
-		connector->force = DRM_FORCE_ON;
+		if (backlight_mode >= CONNECTOR_BACKLIGHT_PWM0 &&
+			backlight_mode <= CONNECTOR_BACKLIGHT_AUX_HDR_BL)
+			connector->force = DRM_FORCE_ON;
 		if (backlight_mode == CONNECTOR_BACKLIGHT_PWM0_BL ||
 			backlight_mode == CONNECTOR_BACKLIGHT_AUX_VESA_BL ||
 			backlight_mode == CONNECTOR_BACKLIGHT_AUX_HDR_BL) {
diff -ruN -x '*.o' -x '*.ko' -x '*.mod*' -x '.tmp*' -x Module.symvers -x modules.order a/fantsrvkm/fantdpu_panel_backlight.c b/fantsrvkm/fantdpu_panel_backlight.c
--- a/fantsrvkm/fantdpu_panel_backlight.c	2026-09-09 17:29:54.464460945 +0800
+++ b/fantsrvkm/fantdpu_panel_backlight.c	2026-09-10 09:08:41.784624418 +0800
@@ -996,6 +996,9 @@
 		return -EINVAL;
 	}
 
+	if (backlight_mode == CONNECTOR_BACKLIGHT_DDCCI)
+		return 0;
+
 	bl = devm_kzalloc(panel->dev, sizeof(*bl), fh2m_hal_get_fant_gfp_kernel());
 	if (!bl) {
 		return -ENOMEM;
