diff -ruN -x '*.o' -x '*.ko' -x '*.mod*' -x '.tmp*' -x Module.symvers -x modules.order a/fantsrvkm/fantdpu_drm_gem.c b/fantsrvkm/fantdpu_drm_gem.c
--- a/fantsrvkm/fantdpu_drm_gem.c	2022-01-01 08:00:00.000000000 +0800
+++ b/fantsrvkm/fantdpu_drm_gem.c	2026-09-09 18:39:03.235771112 +0800
@@ -569,10 +569,12 @@
 
 	list_del(&pvm->list);
 #ifdef GEM_USED_VMALLOC
-	fantdpu_dma_memcpy_for_smallbar_sg(obj->dev->dev, &src, &dst, &len, 1, SYS2GDDR);
+	if (pvm->cpu_write)
+		fantdpu_dma_memcpy_for_smallbar_sg(obj->dev->dev, &src, &dst, &len, 1, SYS2GDDR);
 	fh2m_fant_vfree(pvm->vaddr);
 #else
-	fantdpu_dma_memcpy_for_smallbar(obj->dev->dev, &src, &dst, &len, 1, SYS2GDDR);
+	if (pvm->cpu_write)
+		fantdpu_dma_memcpy_for_smallbar(obj->dev->dev, &src, &dst, &len, 1, SYS2GDDR);
 	fh2m_fant_kfree(pvm->vaddr);
 #endif
 
@@ -685,6 +687,8 @@
 
 	INIT_LIST_HEAD(&pvm->list);
 	mutex_lock(&fantdpu_obj->vm_lock);
+	pvm->cpu_write = !fantdpu_obj->cpu_prep ||
+		fantdpu_obj->cpu_prep_write;
 	list_add_tail(&pvm->list, &fantdpu_obj->vm_head);
 	mutex_unlock(&fantdpu_obj->vm_lock);
 
@@ -2844,6 +2848,8 @@
 	fantdpu_gem_object *fantdpu_obj = NULL;
 	bool write = ! !(args->flags & PDP_GEM_CPU_PREP_WRITE);
 	bool wait = !(args->flags & PDP_GEM_CPU_PREP_NOWAIT);
+	bool cpu_write = write || !(args->flags & PDP_GEM_CPU_PREP_READ);
+	struct gem_vm_list *pvm = NULL;
 	int err = 0;
 
 	if (args->flags & ~(PDP_GEM_CPU_PREP_READ | PDP_GEM_CPU_PREP_WRITE | PDP_GEM_CPU_PREP_NOWAIT)) {
@@ -2878,8 +2884,15 @@
 		if (!dma_resv_test_signaled_rcu(fantdpu_obj->resv, write))
 			err = -EBUSY;
 	}
-	if (!err)
+	if (!err) {
+		mutex_lock(&fantdpu_obj->vm_lock);
+		fantdpu_obj->cpu_prep_write = cpu_write;
 		fantdpu_obj->cpu_prep = true;
+		if (cpu_write)
+			list_for_each_entry(pvm, &fantdpu_obj->vm_head, list)
+				pvm->cpu_write = true;
+		mutex_unlock(&fantdpu_obj->vm_lock);
+	}
 
 exit_unref:
 	drm_gem_object_put(gem_obj);
@@ -2916,7 +2929,10 @@
 		goto exit_unref;
 	}
 
+	mutex_lock(&fantdpu_obj->vm_lock);
 	fantdpu_obj->cpu_prep = false;
+	fantdpu_obj->cpu_prep_write = true;
+	mutex_unlock(&fantdpu_obj->vm_lock);
 
 exit_unref:
 	drm_gem_object_put(gem_obj);
diff -ruN -x '*.o' -x '*.ko' -x '*.mod*' -x '.tmp*' -x Module.symvers -x modules.order a/fantsrvkm/include/fantdpu_drm_gem.h b/fantsrvkm/include/fantdpu_drm_gem.h
--- a/fantsrvkm/include/fantdpu_drm_gem.h	2022-01-01 08:00:00.000000000 +0800
+++ b/fantsrvkm/include/fantdpu_drm_gem.h	2026-09-09 18:38:38.691733422 +0800
@@ -93,6 +93,7 @@
 	unsigned long dev_paddr;
 	void * vaddr;
 	int size;
+	bool cpu_write;
 };
 
 typedef struct fantdpu_gem_object_t {
@@ -132,6 +133,7 @@
 
 	bool is_fbdev_obj;
 	bool cpu_prep;
+	bool cpu_prep_write;
 	bool is_duplicate;
 	int dfd;
 	spinlock_t lock;
